File: 9/task.ex
defmodule Task do

  def vc(count, values) do
    vc(count, values, [], "aouiey")
  end

  defp vc(_, [], result, _) do
    Enum.reverse(result)
  end

  defp vc(count, [head_value|tail_values], result, symbols) do
    vc(count - 1, tail_values, [Kernel.length(for << char  <- head_value >>, String.contains?(symbols, <<char>>), do: true)|result], symbols)
  end

end

File: 9/task.erl
-module(task).

-export([vc/2]).

vc(Count, Values) ->
  vc(Count, Values, [], <<"aouiey">>).

vc(_, [], Result, _) ->
  lists:reverse(Result);
vc(Count, [ValueHead|ValuesTail], Result, Symbols) ->
  vc(Count - 1, ValuesTail, [lists:flatlength([true || <<Char>> <= ValueHead, binary:match(Symbols,<<Char>>) /= nomatch])|Result], Symbols).
